src/mappings/entities/mapIdentities.ts

```diff
@@ -361,3 +371,39 @@ const handleSecondaryKeysAdded = async (
 
   await Promise.all(promises);
 };
+
+const handlePrimaryKeyUpdated = async (
+  blockId: string,
+  eventId,
+  params: Codec[]
+): Promise<void> => {
+  const [rawDid, , newKey] = params;
+  const address = getTextValue(newKey);
+
+  const identity = await getIdentity(getTextValue(rawDid));
+
+  Object.assign(identity, {
+    primaryAccount: address,
+    updatedBlockId: blockId,
+    eventId,
+  });
+
+  await identity.save();
+};
+
+const handleSecondaryKeyLeftIdentity = async (params: Codec[]): Promise<void> => {
+  const [, rawAccount] = params;
+
```

Comment:
Right now when secondary keys are removed, we remove both Account and Permission entry, which now that I think of I'm not sure we should be removing altogether or just updating in a way to showcase that the Account is unlinked?

Reply:
I don't think we should use a bool to track it. Its too hard to document to users imo.

Could we have an `AccountIdentity` history? Instead of deleting the record, move it to a history table instead.

I think the `drop not null` constraint makes sense, so the "natural" queries reflect the chain state.